dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -409,6 +409,8 @@ void ExchangeReceiverBase<RPCContext>::waitAllConnectionDone()
     // In order to ensure the destructions of local tunnels are
     // after the ExchangeReceiver, we need to wait at here.
     waitLocalConnectionDone(lock);
+
+    lock.unlock();
```

Comment:
Can you add some comments to explain why `waitLocalConnectionDone` need lock and `waitAsyncConnectionDone` don't need lock?

Reply:
> Can you add some comments to explain why `waitLocalConnectionDone` need lock and `waitAsyncConnectionDone` don't need lock?

okk

Reply:
> why not
> 
> ```
> {
>     std::unique_lock lock(mu);
>     ...
> }
> waitAsyncConnectionDone();
> ```
> 
> ?

okk